conjure/src/main/java/com/palantir/conjure/cli/ConjureCli.java

```diff
@@ -185,6 +189,19 @@ static void generate(CliConfiguration config) {
             }
         }
 
+        private static SafetyDeclarationRequirements getSafetyRequirements(CliConfiguration config) {
+            if (config.requireSafety() && config.requireSafetyExternalImports()) {
+                return SafetyDeclarationRequirements.EXTERNAL_IMPORTS_REQUIRED;
+            } else if (config.requireSafety() && !config.requireSafetyExternalImports()) {
+                return SafetyDeclarationRequirements.REQUIRED;
+            } else if (!config.requireSafety() && !config.requireSafetyExternalImports()) {
+                return SafetyDeclarationRequirements.ALLOWED;
+            } else {
+                System.err.println("If safety is required for external types, it must be required for all types.");
```

Comment:
Because I'm using two booleans to represent three potential cases (safety allowed, safety required, safety required for external types), there's this awkward 4th case that isn't quite correct. We could avoid this weird fourth case by instead migrating the config to an enum with 3 options. 

I avoided the enum because it would take a lot of upfront work to migrate all the existing conjure configurations to an enum, to say nothing of non-palantir usages that we couldn't excavate. On the other hand, the enum gives us more flexibility, which may be important if we want similar configurability as we expand this feature to cover more kinds of external imports in the future.

An in-between solution is to deserialize the requireSafety() config options into a String instead of a boolean, and then map as following:
true -> REQUIRED
false -> ALLOWED
EXTERNAL_IMPORTS_REQUIRED -> EXTERNAL_IMPORTS_REQUIRED
REQUIRED -> REQUIRED
ALLOWED -> ALLOWED

This would give us future config flexibility and is backwards compatible, but is more counterintuitive than the two above options to me. @carterkozak thoughts?